host/src/FunctionsNetHost/Logger.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Globalization;
+
+namespace FunctionsNetHost
+{
+    internal static class Logger
+    {
+        private static readonly bool IsDebugLevelLogEnabled;
+        private static readonly string LogPrefix;
+
+        static Logger()
+        {
+#if !DEBUG
+            LogPrefix = "LanguageWorkerConsoleLog";
+#else
+            LogPrefix = "";
+#endif
+            string traceLoggingEnabled =
+                Environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsNetHostTrace) ?? "0";
+            IsDebugLevelLogEnabled = traceLoggingEnabled == "1";
+        }
+
+        internal static bool IsDebugLogEnabled => IsDebugLevelLogEnabled;
+
+        /// <summary>
+        /// Logs a debug message if "AZURE_FUNCTIONS_FUNCTIONSNETHOST_TRACE" environment variable value is set to "1"
+        /// For optimal performance (when building the log message needs execute some code (Ex: someObj.ToString() or so),
+        /// consider checking "IsDebugLogEnabled" before calling LogDebug.
+        /// </summary>
+        internal static void LogDebug(string message)
+        {
+            if (IsDebugLevelLogEnabled)
+            {
+                Log(message);
+            }
+        }
+
+        /// <summary>
+        /// Logs a debug message if "AZURE_FUNCTIONS_FUNCTIONSNETHOST_TRACE" environment variable value is set to "1"
+        /// </summary>
+        public static void LogDebug(Func<string> messageProvider)
```

Comment:
If the caller code uses this method, It allocates a delegate. But whatever code the caller wants to execute to produce the message string (Ex: Calling some extra methods) won't be executed If DebugLevelLog is not enabled.

We also expose the property "IsDebugLogEnabled" so that callers can use that on the caller side.

```csharp
if (Logger.IsDebugLogEnabled)
{
    Logger.LogDebug($"Foo {someObject.SomeMethod()");
}
```